src/App.tsx

```diff
@@ -1,14 +1,64 @@
-import React from 'react';
+import React, {
+  useCallback, useMemo, useEffect, useState,
+} from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
+import { Person } from './types/Person';
+import { PeopleList } from './components/PeopleList/PeopleList';
+
+const APPLY_QUERY_DELAY = 1000;
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const onApplyQuery = useCallback(
+    debounce((value: string) => setAppliedQuery(value),
+      APPLY_QUERY_DELAY),
+    [],
+  );
+
+  const filteredPeople = useMemo(() => {
+    setIsDropdownVisible(true);
+
+    return peopleFromServer.filter((person) => person.name.toLowerCase()
+      .includes(appliedQuery.toLowerCase()));
+  }, [appliedQuery]);
+
+  const handleQueryChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(event.target.value);
+      onApplyQuery(event.target.value);
+    },
+    [],
```

Comment:
do not ignore lint hints about dependencies next time:
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/76d515b6-8747-41ea-9b65-a73d1ae8ba6a)